detox/src/server/DetoxServer.js

```diff
@@ -71,8 +77,52 @@ class DetoxServer {
     }
   }
 
-  close() {
+  async close() {
+    if (this._hasConnections()) {
```

Comment:
The heart of the tear-down matter.
Note that the server web-socket's `close()` allows for giving it a callback that is essentially called when closing end. I've tried it - it just doesn't work (called prematurely).